ds701_book/26-TimeSeries.qmd

```diff
@@ -0,0 +1,953 @@
+decomposition.observed.plot(ax=ax1)
+ax1.set_ylabel('Observed')
+decomposition.trend.plot(ax=ax2)
+ax2.set_ylabel('Trend')
+decomposition.seasonal.plot(ax=ax3)
+ax3.set_ylabel('Seasonal')
+decomposition.resid.plot(ax=ax4)
+ax4.set_ylabel('Residual')
+plt.tight_layout()
+plt.show()
+```
+
+
+## STL Decomposition
+
+The STL decomposition is more flexible and can handle the seasonality in the data
+better.
+
+Let's try the STL decomposition:
+
+```{python}
+from statsmodels.tsa.seasonal import STL
+
+stl = STL(ts, period=12, robust=True)
+result = stl.fit()
+# result.plot()
+
+# Plot the decomposed components
+fig, (ax1, ax2, ax3, ax4) = plt.subplots(4, 1, figsize=(8, 6))
+result.observed.plot(ax=ax1)
+ax1.set_ylabel('Observed')
+ax1.set_title('Air Passengers 1949-1960 (STL)')
+result.trend.plot(ax=ax2)
+ax2.set_ylabel('Trend')
+result.seasonal.plot(ax=ax3)
+ax3.set_ylabel('Seasonal')
+result.resid.plot(ax=ax4)
+ax4.set_ylabel('Residual')
+plt.tight_layout()
+plt.show()
+```
+
+
+
+# Stationarity and Differencing
+
+## Definition of stationarity
+
+A time series is said to be stationary if its statistical properties such as mean, variance, and autocorrelation are constant over time.
+
+**Types of Stationarity**:
+
+1. **Strict Stationarity**: The entire distribution of the time series remains unchanged over time.
+2. **Weak Stationarity**: Only the first two moments (mean and variance) are constant over time, and the covariance between two time points depends only on the time lag between them.
+
+## Importance of stationarity in time series analysis
+
+Stationarity is crucial in time series analysis because many statistical methods and models assume that the time series is stationary. Non-stationary data can lead to misleading results and poor forecasts.
+
+**Why Stationarity Matters**:
+
+1. **Model Assumptions**: Many time series models, such as ARIMA, require the data to be stationary to make accurate predictions.
+2. **Statistical Properties**: Stationary time series have consistent statistical properties over time, making it easier to analyze and interpret.
+3. **Forecasting**: Stationary data improves the reliability and accuracy of forecasts.
+
+## Techniques to achieve stationarity
+
+1. **Differencing**: Subtracting the previous observation from the current observation to remove trends and seasonality.
+2. **Transformation**: Applying mathematical transformations such as logarithms or square roots to stabilize the variance.
+3. **Decomposition**: Separating the time series into trend, seasonal, and residual components to analyze and adjust each part individually.
+4. **Smoothing**: Using techniques like moving averages to smooth out short-term fluctuations and highlight longer-term trends.
+
+## Demonstration of stationarity tests
+
+- **ADF Test (Augmented Dickey-Fuller Test)**: A statistical test used to determine if a time series is stationary. It tests the null hypothesis that a unit root is present in the time series.
```

Comment:
What is a unit root?

Reply:
I added a definition as footnote. But maybe we should just get rid of this slide? Because we don't really go into any details? or maybe just say these tests exist but we won't go into them.

Reply:
Ok. I added examples of applying the tests. We can always skip in lecture and leave in course notes.